test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -4215,6 +4228,53 @@ BOOST_AUTO_TEST_CASE(generic_delegatecall)
 	BOOST_CHECK_EQUAL(balanceAt(c_senderAddress), 50 + 11);
 }
 
+BOOST_AUTO_TEST_CASE(generic_staticcall)
+{
+	if (dev::test::Options::get().evmVersion().hasStaticCall())
+	{
+		char const* sourceCode = R"**(
+				contract A {
+					uint public x;
+					constructor() public { x = 42; }
+					function pureFunction(uint256 p) public pure returns (uint256) { return p; }
+					function viewFunction(uint256 p) public view returns (uint256) { return p + x; }
+					function nonpayableFunction(uint256 p) public returns (uint256) { x = p; return x; }
+					function assertFunction(uint256 p) public view returns (uint256) { assert(x == p); return x; }
+				}
+				contract C {
+					function f(address a) public view returns (bool)
+					{
+						bytes4 signature = bytes4(bytes32(keccak256("pureFunction(uint256)")));
+						return a.staticcall(abi.encodeWithSelector(signature, 23));
```

Comment:
Why not use `encodeWithSignature`?

Reply:
No reason - just copied the test from ``delegatecall`` - I'll change it.

Reply:
Done.

Reply:
I left it the same as before for ``delegatecall``, though.